File: src/main/java/io/github/m/MField.java
package io.github.m;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation which marks a field to be accessible from M code.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface MField {
    String name();
}

File: src/main/kotlin/io/github/m/MField.kt
package io.github.m

/**
 * Annotation which marks a field to be accessible from M code.
 *
 * @param name The name of the field in M code.
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class MField(val name: String)